### PR TITLE
Upgrade dependencies (Express 4.0 is biggest one)

### LINKED ISSUE #422: Update dependencies
https://david-dm.org/rendrjs/rendr

Need to update:
* express(*)
* underscore
* request

(*) To do this I believe we need to drop Node 0.8 support.  Anyone have any thoughts on dropping support for 0.8? @spikebrehm @bigethan @lo1tuma @c089 @alexindigo

### PR DESCRIPTION
**Pull request by saponifi3d**
Update dependencies:
- request 2.40.0 -> 2.51.0
- underscore 1.6.0 -> 1.7.0
- express 3 -> 4.11.0

Drop node 0.8 support

Bump version rendr to 1.0.0

Fixes #422 
@spikebrehm

### CODE REVIEW

**Comment by jmerrifield** [comment]
:+1:

File: server/router.js [hunk 1]
```diff
@@ -123,24 +106,3 @@ ServerRouter.prototype.getHeadersForRoute = function(definition) {
   }
   return headers;
 };
-
-/**
- * Return the route definition based on a URL, according to the routes file.
- * This should match the way Express matches routes on the server, and our
- * ClientRouter matches routes on the client.
- */
-ServerRouter.prototype.match = function(pathToMatch) {
```
Comment (spikebrehm) [inline server/router.js]: curious why this is removed

**Comment by spikebrehm** [comment]
exciting!

File: server/router.js (hunk 1 above)
Comment (saponifi3d) [inline server/router.js]: It's no longer in use - I couldn't find any references to it in the code outside of tests.  Everything is working locally on test sites and change.org's main site with it too.

Comment (spikebrehm) [inline server/router.js]: :ship:

**Pull request merged**